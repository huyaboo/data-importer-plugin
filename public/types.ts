import { DataSourceManagementPluginSetup } from 'src/plugins/data_source_management/public';
import { DataSourcePluginSetup } from 'src/plugins/data_source/public';
import { NavigationPublicPluginStart } from '../../../src/plugins/navigation/public';

export interface DataImporterPluginSetupDeps {
  dataSourceManagement?: DataSourceManagementPluginSetup;
  dataSource?: DataSourcePluginSetup;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface DataImporterPluginSetup {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface DataImporterPluginStart {}

export interface DataImporterPluginStartDependencies {
  navigation: NavigationPublicPluginStart;
}

export interface ImportResponse {
  message: {
    total: number;
    message: string;
  };
  success: boolean;
}
