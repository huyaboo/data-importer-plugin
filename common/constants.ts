export const CSV_FILE_TYPE = 'csv';
export const JSON_FILE_TYPE = 'json';
export const NDJSON_FILE_TYPE = 'ndjson';

export const SUPPORTED_FILE_TYPES = [CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE];
