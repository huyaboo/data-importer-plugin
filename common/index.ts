export const PLUGIN_ID = 'staticDataIngestion';
export const PLUGIN_NAME = 'staticDataIngestion';
