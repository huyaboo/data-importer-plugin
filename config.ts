import { schema, TypeOf } from '@osd/config-schema';

export const configSchema = schema.object({
  enabledFileTypes: schema.arrayOf(
    schema.oneOf([schema.literal('csv'), schema.literal('json'), schema.literal('ndjson')]),
    {
      defaultValue: [],
    }
  ),
  maxFileSizeBytes: schema.number({
    defaultValue: 100000000,
  }),
  maxTextCount: schema.number({
    defaultValue: 10000,
  }),
});

export type ConfigSchema = TypeOf<typeof configSchema>;
