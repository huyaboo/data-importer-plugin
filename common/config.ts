import { schema, TypeOf } from '@osd/config-schema';
import { CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE } from './constants';

export const configSchema = schema.object({
  enabled: schema.boolean({ defaultValue: false }),
  enabledFileTypes: schema.arrayOf(schema.string(), {
    defaultValue: [CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE],
  }),
  maxFileSizeBytes: schema.number({
    defaultValue: 100000000,
    min: 1,
  }),
  maxTextCount: schema.number({
    defaultValue: 10000,
    min: 1,
  }),
});

export type ConfigSchema = TypeOf<typeof configSchema>;
export type PublicConfigSchema = Omit<ConfigSchema, 'enabled'>;
