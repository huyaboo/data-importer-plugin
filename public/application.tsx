import React from 'react';
import ReactDOM from 'react-dom';
import { AppMountParameters, CoreStart } from '../../../src/core/public';
import { DataImporterPluginStartDependencies } from './types';
import { DataImporterPluginApp } from './components/app';
import { PublicConfigSchema } from '../config';
import { DataImporterPluginSetupDeps } from './types';

export const renderApp = (
  { notifications, http, savedObjects }: CoreStart,
  { navigation }: DataImporterPluginStartDependencies,
  { appBasePath, element }: AppMountParameters,
  { dataSource, dataSourceManagement }: DataImporterPluginSetupDeps,
  config: PublicConfigSchema
) => {
  ReactDOM.render(
    <DataImporterPluginApp
      basename={appBasePath}
      notifications={notifications}
      http={http}
      navigation={navigation}
      config={config}
      savedObjects={savedObjects}
      dataSourceEnabled={!!dataSource}
      dataSourceManagement={dataSourceManagement}
    />,
    element
  );

  return () => ReactDOM.unmountComponentAtNode(element);
};
