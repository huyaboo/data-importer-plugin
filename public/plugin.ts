import { i18n } from '@osd/i18n';
import {
  AppMountParameters,
  CoreSetup,
  CoreStart,
  Plugin,
  PluginInitializerContext,
} from '../../../src/core/public';
import {
  DataImporterPluginSetup,
  DataImporterPluginStart,
  AppPluginStartDependencies,
} from './types';
import { PLUGIN_NAME, PLUGIN_NAME_AS_TITLE } from '../common';
import { ConfigSchema } from '../config';

export class DataImporterPlugin
  implements Plugin<DataImporterPluginSetup, DataImporterPluginStart> {
  private readonly config: ConfigSchema;

  constructor(initializerContext: PluginInitializerContext<ConfigSchema>) {
    this.config = initializerContext.config.get();
  }

  public setup(core: CoreSetup): DataImporterPluginSetup {
    const config = this.config;
    // Register an application into the side navigation menu
    core.application.register({
      id: 'dataImporterPlugin',
      title: PLUGIN_NAME_AS_TITLE,
      async mount(params: AppMountParameters) {
        // Load application bundle
        const { renderApp } = await import('./application');
        // Get start services as specified in opensearch_dashboards.json
        const [coreStart, depsStart] = await core.getStartServices();
        // Render the application
        return renderApp(coreStart, depsStart as AppPluginStartDependencies, params, config);
      },
    });

    // Return methods that should be available to other plugins
    return {
      getGreeting() {
        return i18n.translate('dataImporterPlugin.greetingText', {
          defaultMessage: 'Hello from {name}!',
          values: {
            name: PLUGIN_NAME,
          },
        });
      },
    };
  }

  public start(core: CoreStart): DataImporterPluginStart {
    return {};
  }

  public stop() {}
}
