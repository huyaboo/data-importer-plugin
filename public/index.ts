import { PluginInitializerContext } from '../../../src/core/public';
import './index.scss';

import { DataImporterPlugin } from './plugin';

// This exports static code and TypeScript types,
// as well as, OpenSearch Dashboards Platform `plugin()` initializer.
export function plugin(initializerContext: PluginInitializerContext) {
  return new DataImporterPlugin(initializerContext);
}
export { DataImporterPluginSetup, DataImporterPluginStart } from './types';
