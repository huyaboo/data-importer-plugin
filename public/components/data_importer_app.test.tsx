import React from 'react';
import { DataImporterPluginApp } from './data_importer_app';
import { coreMock } from '../../../../src/core/public/mocks';
import { testDataSourceManagementPlugin } from '../../../../src/plugins/data_source_management/public/mocks';
import { PublicConfigSchema } from '../../config';
import { navigationPluginMock } from '../../../../src/plugins/navigation/public/mocks';
import { shallow } from 'enzyme';
import { SUPPORTED_FILE_TYPES_LIST } from '../../common';

describe('App', () => {
  const notificationsMock = coreMock.createStart().notifications;
  const httpMock = coreMock.createStart().http;
  const savedObjectsMock = coreMock.createStart().savedObjects;
  const navigationMock = navigationPluginMock.createStartContract();
  const mockConfig: PublicConfigSchema = {
    enabledFileTypes: SUPPORTED_FILE_TYPES_LIST,
    maxFileSizeBytes: 104857600,
    maxTextCount: 10000,
  };

  it('should render without MDS', () => {
    const container = shallow(
      <DataImporterPluginApp
        basename={'dataImporterPlugin'}
        notifications={notificationsMock}
        http={httpMock}
        savedObjects={savedObjectsMock}
        navigation={navigationMock}
        config={mockConfig}
        dataSourceEnabled={false}
        dataSourceManagement={undefined}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with MDS', () => {
    const dataSourceManagementMock = testDataSourceManagementPlugin(
      coreMock.createSetup(),
      coreMock.createStart()
    ).setup;

    // @ts-expect-error
    dataSourceManagementMock.ui.getDataSourceMenu = jest.fn(() => <div>DataSourceMenu</div>);

    const container = shallow(
      <DataImporterPluginApp
        basename={'dataImporterPlugin'}
        notifications={notificationsMock}
        http={httpMock}
        savedObjects={savedObjectsMock}
        navigation={navigationMock}
        config={mockConfig}
        dataSourceEnabled={true}
        dataSourceManagement={dataSourceManagementMock}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
