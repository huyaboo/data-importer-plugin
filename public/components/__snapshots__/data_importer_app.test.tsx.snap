// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render with MDS 1`] = `ShallowWrapper {}`;

exports[`App should render without MDS 1`] = `ShallowWrapper {}`;
